include/inja/parser.hpp

```diff
@@ -621,9 +621,9 @@ class Parser {
                   const FunctionStorage& function_storage)
       : config(parser_config), lexer(lexer_config), template_storage(template_storage), function_storage(function_storage) {}
 
-  Template parse(std::string_view input, std::string_view path) {
-    auto result = Template(static_cast<std::string>(input));
-    parse_into(result, path);
+  std::shared_ptr<Template> parse(std::string_view input, std::string_view path) {
```

Comment:
should we use unique_ptr here? i don't see this shared anywhere?

Reply:
Great callout, thanks! Made that change